src/commonTest/kotlin/org/intellij/markdown/JsonSpecTestBase.kt

```diff
@@ -0,0 +1,63 @@
+package org.intellij.markdown
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import org.intellij.markdown.flavours.MarkdownFlavourDescriptor
+import org.intellij.markdown.html.HtmlGenerator
+import org.intellij.markdown.parser.MarkdownParser
+import kotlin.test.assertEquals
+
+@Serializable
+private data class SpecTest(
+        val markdown: String,
+        val html: String,
+        val example: Int,
+        val start_line: Int,
+        val end_line: Int,
+        val section: String,
+)
+
+abstract class JsonSpecTestBase : TestCase() {
+    abstract fun getTestDataPath(): String
+    abstract fun getFlavour(): MarkdownFlavourDescriptor
+
+    fun doTest(example: Int) {
+        val tests = Json { ignoreUnknownKeys = true }.decodeFromString<List<SpecTest>>(readFromFile(getTestDataPath()))
```

Comment:
The full test suite runs in <200ms on my machine, so I don't think there's any performance issue with reading the entire json file for each test.

Reply:
True, but why not just use `lazy` for the read part? Are there downsides to that that I don't see?